### ISSUE TITLE
Clarification on Metrics in ChatKBQA Results Reproduction

### ISSUE THREAD

**Issue by FUTUREEEEEE** [LHRLAB/ChatKBQA#5]
Hello,

I am attempting to reproduce the results of ChatKBQA on the WebQSP dataset, and I have some confusion regarding the metrics used. Specifically, I am trying to determine which of the provided metrics in the repository corresponds to the "F1 Hits@1 Acc 79.8 83.2 73.8" reported in the paper's results.

In the repository, the following metrics are provided for the WebQSP dataset:
`total:1639, 
                    ex_cnt:1026, 
                    ex_rate:0.6259914582062233, 
                    real_ex_rate:0.6424546023794615, 
                    contains_ex_cnt:1227, 
                    contains_ex_rate:0.74862721171446
                    real_contains_ex_rate:0.7683155917345021`

I would appreciate it if you could help me understand which of these metrics corresponds to the "F1 Hits@1 Acc" reported in the paper. This clarification will greatly assist me in accurately reproducing the results.

Thank you for your assistance.

Best regards,

**Comment by LHRLAB** [comment]
This is just an intermediate result; we still need to evaluate the KBQA result with Retrieval.

**Comment by FUTUREEEEEE** [comment]
Thanks for clearance, after evaluation I got following results, seems match with paper:
`Number of questions: 1639
Average precision over questions: 0.785
Average recall over questions: 0.814
Average f1 over questions (accuracy): 0.783
F1 of average recall and average precision: 0.799
True accuracy (ratio of questions answered exactly correctly): 0.740
Hits@1 over questions: 0.829`

Regards the evaluate process, I found some predicted answers format is unmatched with the label. For example, for question: "WebQTest-362", the predict answer is ['1968-01-01 00:00:00'], while the label are "'m.016j83', 'm.01gqg3', 'm.037pbp', 'm.03kz35', 'm.03r8xj', 'm.03xvj', 'm.03ymyvf', 'm.045...."  

This will lead to a decrease of the final accuracy, right?

![image](https://github.com/LHRLAB/ChatKBQA/assets/52389798/c225068d-b6fe-4a15-bdea-9422adfcba52)

**Comment by LHRLAB** [comment]
It should be. The outcome of the query seems to be time-related, but in reality, it's a collection of entities of the label type.

**Comment by gongchuanyang** [comment]
Continue to evaluate the other entries
The problem WebQTest-2024 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2025 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2027 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2028 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2029 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2030 is not in the prediction set
Continue to evaluate the other entries
The problem WebQTest-2031 is not in the prediction set
Continue to evaluate the other entries
Number of questions: 1639
Average precision over questions: 0.000
Average recall over questions: 0.000
Average f1 over questions (accuracy): 0.000
测试过程中出现上述的结果，是Virtuoso 没成功开启吗？

**Comment by WangYQ999** [comment]
Hello, I have a question about the difference between "Train LLMs for Logical Form Generation" and "Beam-setting LLMs for Logical Form Generation"when fine-tuning the large model，can I make ft directly in Beam search.

**Comment by LHRLAB** [comment]
Training LLMs for Logical Form Generation is the SFT phase, which is the training phase. Beam-setting LLMs for Logical Form Generation uses the SFT-trained model to perform beam search generation, which is not a training phase.

**Comment by ZepengDu** [comment]
作者您好，请问您用的微调框架是哪个？是否支持预训练呢？
